src/Examine/LuceneEngine/Directories/ExamineDirectory.cs

```diff
@@ -15,6 +15,12 @@ public virtual void InvokeOnCommit(ExamineIndexWriter writer)
             _onCommitAction?.Invoke(writer);
         }
 
+        public event EventHandler HandleOutOfSyncEvent;
+
+        public void HandleOutOfSyncDirectory()
+        {
+            HandleOutOfSyncEvent?.Invoke(this, null);
```

Comment:
We could avoid this class using this pr https://github.com/Shazwazza/Examine/pull/210 as examineDirectory mostly exists to set up the writer. Overriding the lucenceIndex WriterFactory seems to be a cleaner approach

Reply:
@nzdev looks promising, not sure what Shannon is thinking there let see :)